tools/cfn2ts/lib/codegen.ts

```diff
@@ -586,10 +607,19 @@ export default class CodeGenerator {
   }
 
   private renderTypeUnion(context: genspec.CodeName, types: genspec.CodeName[]) {
-    return types.map((type) => type.relativeTo(context).fqn).join(' | ');
+    return types
+      .map(type => type.relativeTo(context))
+      .map(type => {
+        const specType = type.specName && this.spec.PropertyTypes[type.specName.fqn];
+        if (!specType || schema.isPropertyBag(specType)) {
+          return type.fqn;
+        }
+        return this.findNativeType(context, specType);
+      })
+      .join(' | ');
   }
 
-  private docLink(link: string | undefined, ...before: string[]) {
+  private docLink(link: string | undefined, ...before: string[]) {
```

Comment:
Did you replace one kind of a space with another one here? WTF is this diff?

Reply:
Yes. Non-breaking space versus regular space. I see them because I have the [`highlight-bad-chars` plug-in](https://marketplace.visualstudio.com/items?itemName=wengerk.highlight-bad-chars) installed.